javacutil/src/org/checkerframework/javacutil/TypesUtils.java

```diff
@@ -163,6 +163,21 @@ public static boolean isPrimitive(TypeMirror type) {
         }
     }
 
+    /**
+     * Returns true iff the arguments are both the same primitive types.
+     *
+     * <p>This is needed because class {@code Type.ClassType} does not override equals.
+     *
+     * @return whether the arguments are the same primitive types
+     */
+    public static boolean areSameDeclaredTypes(Type.ClassType t1, Type.ClassType t2) {
+        // Do a cheaper test first
+        if (!t1.tsym.name.toString().equals(t2.tsym.name.toString())) {
```

Comment:
Couldn't you further optimise this to `if (t1.tsym.name != t2.tsym.name)`?
The two `Name` instances are comparable by reference equality, as they are created by the same compiler instance.